networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AggregateGossipManager.java

```diff
@@ -53,7 +51,8 @@ public void onNewAggregate(final ValidateableAttestation validateableAttestation
     if (!validateableAttestation.isAggregate() || !validateableAttestation.markGossiped()) {
       return;
     }
-    validator.addSeenAggregate(validateableAttestation);
+    // TODO: implement adding the on new aggregate to the validator
+    //    validator.addSeenAggregate(validateableAttestation);
```

Comment:
I don't think we can afford to leave this TODO undone as we'd wind up validating more aggregates than we need to.

Reply:
I believe we had put this here in order to make sure we don’t process aggregates that we’ve created ourselves when they do come back at us. Thinking out loud, there should probably be a check for this at a lower level. I’m guessing jvm-libp2p does not deliver messages that we’ve created back to us. 

Nevertheless, in case it doesn’t, I added the functionality again. This time, after we process attestations at the AttestationManager, for in-house produced attestations, we call attestation and aggregate validators methods to mark them as seen. I saw no reason for calling .addSeen for aggregates and not for single attestations so I added that as well.